test/server/cards/01_SOR/units/LieutenantChildsenDeathStarPrisonWarden.spec.ts

```diff
@@ -0,0 +1,153 @@
+        describe('Lieutenant Childsen\'s When Played ability', function() {
+            describe('when hand contains some Vigilance cards', function() {
+                beforeEach(function () {
+                    contextRef.setupTest({
+                        phase: 'action',
+                        player1: {
+                            hand: [
+                                'lieutenant-childsen#death-star-prison-warden',
+                                'pyke-sentinel',
+                                'system-patrol-craft',
+                                'cargo-juggernaut',
+                                'resilient',
+                                'vanquish',
+                                'confiscate',
+                                'underworld-thug',
+                            ],
+                            groundArena: ['atst'],
+                            spaceArena: ['cartel-spacer'],
+                        },
+                        player2: {
+                            hand: ['rivals-fall'],
+                            groundArena: ['wampa'],
+                            spaceArena: ['alliance-xwing']
+                        }
+                    });
+                });
+
+                it('should gain experience based on the number of revelead cards', function () {
+                    const { context } = contextRef;
+
+                    const reset = () => {
+                        context.player2.resources.forEach((card) => card.exhausted = false);
+                        context.player2.moveCard(context.rivalsFall, 'hand');
+                        context.player2.clickCard(context.rivalsFall);
+                        context.player2.clickCard(context.lieutenantChildsen);
+                        context.player1.moveCard(context.lieutenantChildsen, 'hand');
+                        context.player1.resources.forEach((card) => card.exhausted = false);
+                    };
+
+                    // Reveal 1 Vigilance card
+                    context.player1.clickCard(context.lieutenantChildsen);
+                    expect(context.player1).toBeAbleToSelectExactly([context.pykeSentinel, context.systemPatrolCraft, context.cargoJuggernaut, context.resilient, context.vanquish]);
+                    context.player1.clickCard(context.pykeSentinel);
+                    context.player1.clickPrompt('Done');
+
+                    expect(context.getChatLogs(2)).toEqual([
+                        'player1 uses Lieutenant Childsen to reveal a card',
+                        'player1 reveals Pyke Sentinel due to Lieutenant Childsen',
+                    ]);
+
+                    expect(context.lieutenantChildsen).toHaveExactUpgradeNames(['experience']);
+                    expect(context.player2).toBeActivePlayer();
+
+                    reset();
+
+                    // Reveal 2 Vigilance cards
+                    context.player1.clickCard(context.lieutenantChildsen);
```

Comment:
We can simplify this test by just checking some key points in terms of number of cards - 0, 1, 4